diff --git a/vendor/magento/module-webapi-async/Controller/Rest/Asynchronous/InputParamsResolver.php b/vendor/magento/module-webapi-async/Controller/Rest/Asynchronous/InputParamsResolver.php
--- a/vendor/magento/module-webapi-async/Controller/Rest/Asynchronous/InputParamsResolver.php
+++ b/vendor/magento/module-webapi-async/Controller/Rest/Asynchronous/InputParamsResolver.php
@@ -119,11 +119,13 @@
         $this->inputArraySizeLimitValue->set($route->getInputArraySizeLimit());

         foreach ($this->getInputData() as $key => $singleEntityParams) {
-            $webapiResolvedParams[$key] = $this->resolveBulkItemParams(
-                $singleEntityParams,
-                $routeServiceClass,
-                $routeServiceMethod
-            );
+            if (is_array($singleEntityParams)) {
+                $webapiResolvedParams[$key] = $this->resolveBulkItemParams(
+                    $singleEntityParams,
+                    $routeServiceClass,
+                    $routeServiceMethod
+                );
+            }
         }

         return $webapiResolvedParams;
